mocca-client/src/main/java/com/paypal/mocca/client/MoccaClient.java

```diff
@@ -372,14 +375,18 @@ private AsyncBuilder(final String serverBaseUrl) {
                 }
 
                 public <C extends MoccaClient> C build(final Class<C> apiType) {
+                    MoccaFeignEncoder encoder = new MoccaFeignEncoder();
+
                     AsyncFeign.AsyncBuilder<CC> builder = AsyncFeign.<CC>asyncBuilder()
                         .contract(new MoccaFeignContract())
-                        .encoder(new MoccaFeignEncoder())
+                        .encoder(encoder)
                         .decoder(new MoccaFeignDecoder());
                     if (asyncClient != null) {
                         builder = builder.client(asyncClient);
                     }
-                    return builder.target(apiType, serverUrl);
+                    C client =  builder.target(apiType, graphQLUrlString);
+                    encoder.setClient(client);
```

Comment:
Let me state here that I don't like this change at all. Because this is so tortured I think there must be a better way to do this. But the reason we add the client here is that the Validator class requires the object whose method is being invoked to run.

Reply:
Ok. Let's discuss this in our meeting on Friday.